src/charm.py

```diff
@@ -94,6 +95,15 @@ def restart_agent_service(self) -> None:
 
         self.model.unit.status = ops.ActiveStatus()
 
+    def _on_update_status(self, _: ops.UpdateStatusEvent) -> None:
+        """Update status event hook.
+
+        Raises:
+            RuntimeError: when the service is not running.
+        """
+        if not self.jenkins_agent_service.is_active:
+            raise RuntimeError("jenkins-agent service is not running")
```

Comment:
should we log something perhaps from the service logs to assist the operator with fixing the issue?

Reply:
It seems that `juju-log` does not work very well when logs are pulled directly from journalctl since it's usually need to be large string to have any meaningful information (ref: https://github.com/canonical/operator/issues/531). So I'd say we keep this PR as it is for now while we think of a more suitable logging implementation.